.github/workflows/ci.yml

```diff
@@ -0,0 +1,109 @@
+name: continuous-integration
```

Comment:
Wow, this is long.  Does it have to be this complicated?

Reply:
It might be a bit more verbose, but I don't think it is complicated. Note that this now also includes the `pre-commit` step which didn't exist in the travis configuration. I also explicitly split them in separate jobs and do not run everything sequentially in a single job. This gives more granularity to inspect the results and with GHA we have enough throughput that having them in individual jobs does not negatively effect the performance